sw/device/tests/consecutive_irqs/consecutive_irqs_test.c

```diff
@@ -0,0 +1,292 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/common.h"
+#include "sw/device/lib/dif/dif_plic.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/handler.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/ibex.h"
+
+#define UART0_BASE_ADDR 0x40000000u
+#define PLIC0_BASE_ADDR 0x40090000u
+
+#define PLIC_TARGET kDifPlicTargetIbex0
+
+#define PLIC_PRIORITY_MIN 0u
+#define PLIC_PRIORITY_MAX 3u
+
+static dif_plic_t plic0;
+static dif_uart_t uart0;
+
+// These flags are used in the test routine to verify that a corresponding
+// interrupt has elapsed, and has been serviced.
+static bool uart_rx_overflow_handled = false;
+static bool uart_tx_empty_handled = false;
+static bool uart_handled_more_than_once = false;
+
+/**
+ * Polled implementation of the string send operation.
+ */
+static void uart_send_string(const char *string) {
```

Comment:
Could this be used with print.h, or is the fact that this is polling incompatible?

Reply:
`print.h` doesn't know anything about UART, you have to set up a specific callback for it; how it calls UART is up to you.